### DISCUSSION TITLE
Which models are suitable for causal A/B

### DISCUSSION THREAD

**Discussion by JustM57** [zalandoresearch/pytorch-ts#132] [Q&A]
I would like to test some probabilistic models from this repository for causal A/B. The idea is similar to causal impact where we have mcmc samples and the sum(or other aggregation) to get the results on the evaluation period. After aggregation we have a confidence interval for the whole period(not only single point).
For example, Unobserved Components from statsmmodels or fbprohet same this functionality.
I was looking forward to try something else.  Maybe MAF transformer or TimeGrad?..

**Comment by kashif** [comment]
so most of the models here are conditional probabilistic models where one learns the prob. of the next time step given the past and covariates etc. When the covariates are not causally connected to the target (and of course known in the future for dynamic covariates) all is good and one can sample from the learned distribution given different covariates.

However, when the covariates are causally connected the use of such conditional probabilistic models is problematic. The Causal-deep-AR model offers one solution for the case of a single causal covariate and target variable in the univariate setting.  At inference time one can then use the "do" operator or sample from the covariate head if it is not known at inference time. Hopefully, that gives you an idea?
